shared/rakefiles/xk_config.rake

```diff
@@ -38,12 +50,15 @@ rule @kubectl_creds_file => [@gcp_creds_file] do
   # This duplicates information in terraform code, 'zone'. Could be a variable with some plumbing.
   zone = 'us-central1-a'
   sh "
-    existing_cluster=$(gcloud container clusters list --filter #{cluster_name} --zone #{zone} --project #{ENV["TF_VAR_project_id"]})
+    existing_cluster=$(gcloud container clusters list --filter #{cluster_name} --zone #{zone} --project $TF_VAR_project_id)
     if [ $? == 0 ] && [ \"$existing_cluster\" != \"\" ]; then \
-      gcloud container clusters get-credentials #{cluster_name} --zone #{zone} --project #{ENV["TF_VAR_project_id"]}
+      gcloud container clusters get-credentials #{cluster_name} --zone #{zone} --project $TF_VAR_project_id
     fi"
 end
 
-task :configure_current_project => [@gcp_creds_file] do
+task :configure => [@gcp_creds_file, @default_creds_file, @kubectl_creds_file] do
   sh "gcloud config set project $TF_VAR_project_id"
+  # We need to set service account key var only if SA credentials are present locally.
+  # In case var is unset, application-default credentials will be used.
+  ENV['TF_VAR_serviceaccount_key'] = File.file?(@serviceaccount_key_file) ? @serviceaccount_key_file : ''
```

Comment:
Please change all these single quotes to double quotes.

Reply:
Done.